openrndr-math/src/main/kotlin/org/openrndr/math/Vector2.kt

```diff
@@ -46,6 +46,14 @@ data class Vector2(val x: Double, val y: Double) : Serializable {
         val UNIT_X = Vector2(1.0, 0.0)
         val UNIT_Y = Vector2(0.0, 1.0)
         val INFINITY = Vector2(Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY)
+
+        fun fromAngle(angle: Double): Vector2 {
+            val theta = toRadians(angle)
+            val x = cos(theta)
+            val y = sin(theta)
+
+            return Vector2(x, y).normalized
```

Comment:
normalization is not needed here, at this point x and y already lie on a unit circle

Reply:
Just throwing here an alternative: some frameworks have something like

    vec2 xy = toCartesian(angle, distance);
    vec2 angleDist = toPolar(x, y);

and `fromAngle(a)` would be `toCartesian(a, 1.0);`

I like being able to go both ways, not just to cartesian. Spherical is also useful.

I find naming the hardest part. Possible names: `fromAngle, toAngle, toCartesian, fromCartesian, toPolar, fromPolar, toSpherical, fromSpherical`...

Reply:
@edwinRNDR 🤦‍♂

@hamoid "There are only two hard things in Computer Science: cache invalidation and naming things." -- Phil Karlton

Hate to throw more into the mix, but what about `getDirection` or `getHeading`?

Reply:
@ricardomatias Those are also good if the length is assumed to be 1.0, or as an extra method when you don't care about length.

Reply:
@hamoid I assumed you always want to have it normalized before measuring the angle?

Reply:
My point above was that sometimes you want to be able to go from cartesian to polar and back, without normalizing, because that would destroy some info. So

    vec2 xy = toCartesian(angle : 0, distance : 2); // pseudocode

would give you a `vec2(2, 0);` representing x and y, and 

    vec2 angleDist = toPolar(1, 1);

would give you `vec2(45deg, 1.41);` representing angle and length.

I find it useful and I've used it with toxiclibs. But I can see people saying that it's confusing that a vec2 can hold both x & y or angle & magnitude.

There are similar methods for spherical coordinates. So you could throw in a point in space and it would return two angles and a magnitude. Or the other way around: you put in magnitude and two angles and it gives you a point in space.

Reply:
@hamoid Thanks for the example, got you. I think it's a good idea. Feel free to add a commit on top of the PR. I guess in the meanwhile (if @edwin is onboard) we'll find some suitable naming.

Reply:
What about introducing Polar, which would be analogous to the [Spherical](https://github.com/openrndr/openrndr/blob/master/openrndr-math/src/main/kotlin/org/openrndr/math/Spherical.kt) class? I found the Vector3/Spherical pair to be quite handy to convert between the two representations.

The fromAngle function, perhaps with a radius argument with a default value of 1.0 seems like a handy short-hand for Vector2.fromPolar(Polar(a,r)) or Polar(a,r).cartesian.

Reply:
@edwinRNDR @hamoid I've added the Polar class.